modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/shared/components/toolbar/UpperToolbar.js

```diff
@@ -60,118 +71,233 @@ export default function UpperToolbar({
 		}
 	};
 
+	const saveDefinition = () => {
+		const successMessage = Liferay.Language.get('workflow-saved');
+
+		setAlertMessage(successMessage);
+
+		let content;
+
+		if (currentEditor) {
+			content = currentEditor.getData();
+		}
+		else {
+			content = serializeDefinition(
+				xmlNamespace,
+				{
+					description: '',
+					name: definitionTitle,
+					version,
+				},
+				elements.filter(isNode),
+				elements.filter(isEdge)
+			);
+		}
+
+		fetch(`${baseURL}/workflow-definitions/save`, {
```

Comment:
@naraandrade What do you think about single responsability in this function? Maybe this inside a 'saveDefinitionService' function (or any other other name) is more readable.

Reply:
@ramonmsp Thanks for pointing this out :) I'll do it 👍🏻